x-pack/plugins/apm/public/components/app/service_groups/service_group_save/select_services.tsx

```diff
@@ -114,88 +114,85 @@ export function SelectServices({
     <Container>
```

Comment:
This change broke a bit the UI.  
<img width="605" alt="image" src="https://user-images.githubusercontent.com/3369346/214547803-15b413da-2852-4377-9b4d-6760c0d18a22.png">


@cee-chen  it seems we need to wrap what was previously in `EuiModalHeaderTitle` inside the `EuiModalBody`  

```suggestion
    <Container>
      <EuiModalHeader>
        <EuiModalHeaderTitle>
          {i18n.translate('xpack.apm.serviceGroups.selectServicesForm.title', {
            defaultMessage: 'Select services',
          })}
        </EuiModalHeaderTitle>
      </EuiModalHeader>

      <EuiModalBody
        style={{
          height: `calc(75vh - ${MODAL_HEADER_HEIGHT}px - ${MODAL_FOOTER_HEIGHT}px)`,
          maxHeight:
            MAX_CONTAINER_HEIGHT - MODAL_HEADER_HEIGHT - MODAL_FOOTER_HEIGHT,
        }}
      >
        <EuiText color="subdued" size="s">
          {i18n.translate(
            'xpack.apm.serviceGroups.selectServicesForm.subtitle',
            {
              defaultMessage:
                'Use a query to select services for this group. The preview shows services that match this query within the last 24 hours.',
            }
          )}
        </EuiText>
        {kueryValidationMessage && (
          <EuiText color="danger" size="s">
            {kueryValidationMessage}
          </EuiText>
        )}
        <EuiSpacer size="s" />
        <EuiFlexGroup gutterSize="s">
          <EuiFlexItem>
            <KueryBar
              placeholder={i18n.translate(
                'xpack.apm.serviceGroups.selectServicesForm.kql',
                { defaultMessage: 'E.g. labels.team: "web"' }
              )}
              onSubmit={(value) => {
                setKuery(value);
              }}
              onChange={(value) => {
                setStagedKuery(value);
              }}
              value={kuery}
              suggestionFilter={(querySuggestion) => {
                if ('field' in querySuggestion) {
                  const {
                    field: {
                      spec: { name: fieldName },
                    },
                  } = querySuggestion;

                  return isSupportedField(fieldName);
                }
                return true;
              }}
            />
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButton
              onClick={() => {
                setKuery(stagedKuery);
              }}
              iconType={!kuery ? 'search' : 'refresh'}
              isDisabled={isServiceListPreviewLoading || !stagedKuery}
            >
              {!kuery
                ? i18n.translate(
                    'xpack.apm.serviceGroups.selectServicesForm.preview',
                    { defaultMessage: 'Preview' }
                  )
                : i18n.translate(
                    'xpack.apm.serviceGroups.selectServicesForm.refresh',
                    { defaultMessage: 'Refresh' }
                  )}
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
        {kuery && data?.items && (
          <EuiText color="success" size="s">
            {i18n.translate(
              'xpack.apm.serviceGroups.selectServicesForm.matchingServiceCount',
              {
                defaultMessage:
                  '{servicesCount} {servicesCount, plural, =0 {services} one {service} other {services}} match the query',
                values: { servicesCount: data?.items.length },
              }
            )}
          </EuiText>
        )}
        <EuiSpacer size="s" />
        <EuiFlexGroup
          direction="column"
          gutterSize="s"
          style={{ height: '100%' }}
        >
          <EuiFlexItem>
            <EuiPanel hasShadow={false} hasBorder paddingSize="s">
              {!kuery && (
                <CentralizedContainer>
                  <EuiText size="s" color="subdued">
                    {i18n.translate(
                      'xpack.apm.serviceGroups.selectServicesForm.panelLabel',
                      { defaultMessage: 'Enter a query to select services' }
                    )}
                  </EuiText>
                </CentralizedContainer>
              )}
              {!data && isServiceListPreviewLoading && (
                <CentralizedContainer>
                  <EuiLoadingSpinner />
                </CentralizedContainer>
              )}
              {kuery && data && (
                <ServiceListPreview
                  items={data.items}
                  isLoading={isServiceListPreviewLoading}
                />
              )}
            </EuiPanel>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiModalBody>
      <EuiModalFooter>
        <EuiFlexGroup>
          <EuiFlexItem>
            <div>
              <EuiButton
                color="text"
                onClick={onEditGroupDetailsClick}
                iconType="sortLeft"
                isDisabled={isLoading}
              >
                {i18n.translate(
                  'xpack.apm.serviceGroups.selectServicesForm.editGroupDetails',
                  { defaultMessage: 'Edit group details' }
                )}
              </EuiButton>
            </div>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButtonEmpty onClick={onCloseModal} isDisabled={isLoading}>
              {i18n.translate(
                'xpack.apm.serviceGroups.selectServicesForm.cancel',
                {
                  defaultMessage: 'Cancel',
                }
              )}
            </EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButton
              fill
              onClick={() => {
                onSaveClick({ ...serviceGroup, kuery });
              }}
              isDisabled={isLoading || !kuery}
            >
              {i18n.translate(
                'xpack.apm.serviceGroups.selectServicesForm.saveGroup',
                { defaultMessage: 'Save group' }
              )}
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiModalFooter>
    </Container>
```

Reply:
Apologies for not QAing this closely enough @kpatticha! Unfortunately the proposed code change wasn't quite working as on prod:

<img width="673" alt="" src="https://user-images.githubusercontent.com/549407/214646796-9406241d-195e-4319-8f5b-d28eb0cb032e.png">

So I went with [another workaround](https://github.com/elastic/kibana/pull/149457/commits/18ad40dd51f4c5fe38aa1594e7910789e893edcc?diff=split&w=1) that involved adding a plain `<div>` wrapper around the header contents. This overrides the flex behavior of `EuiModalHeader` while still allowing `EuiModalBody` to correctly shrink or expand responsively (depending on screen height).

<img width="664" alt="" src="https://user-images.githubusercontent.com/549407/214646855-f7d7a169-0ba4-425f-903f-c4733ab187e1.png">

Reply:
@cee-chen no worries! 

Your workaround looks way better, thanks for doing it :D